rotkehlchen/chain/ethereum/oracles/uniswap.py

```diff
@@ -0,0 +1,405 @@
+        link_asset = None
+        found_first_link, found_second_link = False, False
+        for asset in self.routing_assets:
+            if asset != from_asset:
+                found_first_link = self._find_pool_for(
+                    asset=from_asset,
+                    link_asset=asset,
+                    path=output,
+                )
+                if found_first_link:
+                    link_asset = asset
+                    found_second_link = self._find_pool_for(
+                        asset=to_asset,
+                        link_asset=link_asset,
+                        path=output,
+                    )
+                    if found_second_link:
+                        return output
+
+        if not found_first_link:
+            return []
+
+        # if we reach this point it means that we need 2 more jumps
+        # from asset <1st link> glue asset A <2nd link> glue asset B <3rd link> to asset
+        # find now the part for glue asset B <3rd link> to asset
+        second_link_asset = None
+        for asset in self.routing_assets:
+            if asset != to_asset:
+                found_second_link = self._find_pool_for(
+                    asset=to_asset,
+                    link_asset=asset,
+                    path=output,
+                )
+                if found_second_link:
+                    second_link_asset = asset
+                    break
+
+        if not found_second_link:
+            return []
+
+        # finally find the step of glue asset A <2nd link> glue asset B
+        assert second_link_asset is not None
+        assert link_asset is not None
+        pools = self.get_pool(link_asset, second_link_asset)
+        for pool in pools:
+            if pool != ZERO_ADDRESS:
+                output.insert(1, pool)
+                return output
+
+        return []
+
+    def get_price(
+        self,
+        from_asset: Asset,
+        to_asset: Asset,
+        block_identifier: BlockIdentifier,
+    ) -> Price:
+        """
+        Return the price of from_asset to to_asset at the block block_identifier.
+        External oracles are used if non eth tokens are used.
+
+        Can raise:
+        - PriceQueryUnsupportedAsset
+        - RemoteError
+        """
+        log.debug(
+            f'Searching price for {from_asset} to {to_asset} at '
+            f'{block_identifier!r} with {self.get_oracle_name()}',
+        )
+
+        # Uniswap V2 and V3 use in their contracts WETH instead of ETH
+        if from_asset == A_ETH:
+            from_asset = A_WETH
+        if to_asset == A_ETH:
+            to_asset = A_WETH
+
+        if from_asset == to_asset:
+            return Price(ONE)
+
+        if not (from_asset.is_eth_token() and to_asset.is_eth_token()):
+            raise PriceQueryUnsupportedAsset(
+                f'Neither {from_asset} nor {to_asset} are ethereum tokens for the uniswap oracle',
```

Comment:
The previous error was saying that both are not tokens.

```suggestion
                f'Eeither {from_asset} or {to_asset} are not ethereum tokens for the uniswap oracle',
```